test/libsolidity/SolidityNatspecJSON.cpp

```diff
@@ -2022,6 +2023,49 @@ BOOST_AUTO_TEST_CASE(dev_explicit_inehrit_complex)
 	);
 }
 
+BOOST_AUTO_TEST_CASE(dev_different_return_name)
+{
+	char const *sourceCode = R"(
+		contract A {
+			/// @return y value
+			function g(int x) public pure virtual returns (int y) { return x; }
+		}
+
+		contract B is A {
+			function g(int x) public pure override returns (int z) { return x; }
+		}
+	)";
+
+	char const *natspec = R"ABCDEF({
+		"methods":
+		{
+			"g(int256)":
+			{
+				"returns":
+				{
+					"y": "value"
+				}
+			}
+		}
+	})ABCDEF";
+
+	char const *natspec2 = R"ABCDEF({
+		"methods":
+		{
+			"g(int256)":
+			{
+				"returns":
+				{
+					"_0": "y value"
```

Comment:
Is this how we want to do it? Maybe we shouldn't inherit the `returns` tag in such cases. If we do, then it would make sense to replace "y" by the name of the current return value.

Reply:
well, that would be the default behavior if there is no name given, it made sense to me to default to that if there is a conflict in the parameter names..
Replacing was also an idea I had, but that's not as easy as it maybe sounds with how the code is currently structured.

Reply:
Oh, I see what you mean. Maybe a better alternative would be to complete skip it then.

Reply:
> Replacing was also an idea I had, but that's not as easy as it maybe sounds with how the code is currently structured.

I agree. The current Natspec code needs an overhaul.

Reply:
I think this line should say
```suggestion
					"z": "value"
```

Reply:
If we only have one parameter, correcting it can be done, but if we have several, how should we proceed? Use the order of the parameters to see which one should be which? Skip the whole thing?

Reply:
There should be a proper correspondence between the parameters, isn't there?

Reply:
if we use the order, yes..